packages/core/src/providers/network/connector/connectorController.ts

```diff
@@ -0,0 +1,32 @@
+import { Connector } from "./connector";
+
+export class ConnectorController {
```

Comment:
should handle the lifecycle of the connector including chainID, accounts, block number etc.

Should have safeguards to protect from bugs in the connector implementation